### PR TITLE
Issue 1264 bugfix refactor

### PR DESCRIPTION
**Pull request by archeuclid**
Hello

Added AshStartScriptPlugin with associated keys, templates, tests. Made it so it does not depend on BashStartScriptPlugin keys and defines. It has to be explicitly enabled with `enablePlugins`.

The generated scripts do not have quite the same `usage` as bash/bat scripts. Maybe someone can realign them later and add to docs.

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @archeuclid,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: src/main/scala/com/typesafe/sbt/packager/archetypes/scripts/AshScriptPlugin.scala [hunk 1]
```diff
@@ -69,9 +69,11 @@ import sbt._
   *  enablePlugins(AshScriptPlugin)
   * }}}
   */
+@deprecated("Use AshStartScriptPlugin", "1.4.1")
```
Comment (muuki88) [inline src/main/scala/com/typesafe/sbt/packager/archetypes/scripts/AshScriptPlugin.scala]: I wonder why we need to do this. As mentioned in https://github.com/sbt/sbt-native-packager/issues/1264#issuecomment-548263952 I'm currently not seeing the benefit as we copy all settings, which are identical, but add the complexity of having to deal with versions of a script where actually only one should exist.

File: src/main/resources/com/typesafe/sbt/packager/archetypes/scripts/ash-template [hunk 2]
```diff
@@ -1,105 +1,155 @@
 #!/bin/sh
```
Comment (muuki88) [inline src/main/resources/com/typesafe/sbt/packager/archetypes/scripts/ash-template]: I really like the cleaned up version of this scripts. This is by far more readable and predictable

File: src/main/scala/com/typesafe/sbt/packager/archetypes/scripts/AshScriptPlugin.scala (hunk 1 above)
Comment (archeuclid) [reply src/main/scala/com/typesafe/sbt/packager/archetypes/scripts/AshScriptPlugin.scala]: I just wanted to rename so it is ashSTARTSCRIPTplugin not just ashSCRIPTplugin. I agree we can reuse keys, will change that.

Comment (muuki88) [reply src/main/scala/com/typesafe/sbt/packager/archetypes/scripts/AshScriptPlugin.scala]: That makes sense :+1:

**Comment by archeuclid** [comment]
Hello Ok
This project need lots of refactor.

**Comment by briantopping** [comment]
Is there any way to use what's here?

**Comment by muuki88** [comment]
@briantopping unfortunately not. We have no releases for pull requests. You will need to checkout this pull request and build on your own :( Or you finish this PR and we release :)

**Comment by briantopping** [comment]
No worries, I converted the code that depends on this to pass arguments as environment variables. My shell skills are poor at best.

**Comment by muuki88** [comment]
> My shell skills are poor at best

You are in good company @briantopping Since kubernetes we have to fight more with YAML than anything else :joy:

**Pull request closed without merging**